.devcontainer/Dockerfile

```diff
@@ -1,20 +1,31 @@
-ARG SWIFT_VERSION=5.9
+ARG SWIFT_VERSION=6.1
 # Other ARG declarations must follow FROM
 FROM swift:${SWIFT_VERSION}
 
 ARG HYLO_LLVM_BUILD_TYPE=MinSizeRel
 ARG HYLO_LLVM_BUILD_RELEASE=20240303-215025
 ARG HYLO_LLVM_VERSION=17.0.6
+ARG CMAKE_VERSION=3.29.1
 
 ENV HYLO_LLVM_DOWNLOAD_URL="https://github.com/hylo-lang/llvm-build/releases/download"
 
 RUN apt install -y gnupg
 RUN apt update
 RUN apt install -y curl libzstd-dev libzstd1 lsb-release make ninja-build tar wget zstd software-properties-common python3-pip
 
-# Get a recent cmake (https://www.kitware.com//cmake-python-wheels/)
-RUN if $(/usr/bin/which cmake) ; then apt purge --auto-remove cmake ; fi
-RUN pip3 install --upgrade cmake
+# Remove system cmake if installed
```

Comment:
Why are we replacing a simple method of upgrading cmake with a more complex one?
The original was an officially supported method as referenced in the URL.

Reply:
Hylo's `CMakeLists.txt` says: `cmake_minimum_required(VERSION 3.30)`.  I think we should use Swifty-LLVM as a testbed for Hylo CI/configuration, and try to replicate as much as possible in both places.  So I think `pip` is the better choice.

Reply:
For various reasons I did this upgrade on my local hylo devcontainer; it's not 100% complete, but you might find these changes useful

[Swift-6.1.patch](https://github.com/user-attachments/files/20257593/Swift-6.1.patch)